### ISSUE TITLE
New Azure subscription, fails to refresh access token

### ISSUE THREAD

**Issue by douglasstarnes** [microsoft/vscode-azureresourcegroups#681]
I have a new Azure subscription and I want to demo it using GitHub Codespaces.  However, when I try to sign in with the Azure Resources extension, the only item in the sidebar bar is "Select Subscription".  If I click on that, an error shows in the popup that says "Failed to refresh access token".  However, if I try to access the same subscription using Visual Studio Code, it works.

**Comment by savannahostrowski** [comment]
I also just ran into this issue in Codespaces

![image](https://github.com/microsoft/vscode-azureresourcegroups/assets/8949415/afd88d82-c01d-4edc-a5e2-9caf28183505)

**Comment by alexweininger** [comment]
@douglasstarnes  Just to clarify, are you using GitHub CodeSpaces or github.dev or vscode.dev?

Does reloading the window help at all?

**Comment by douglasstarnes** [comment]
@alexweininger Nope.  I've tried reinstalling the extension, logging out and back in, and creating a new Codespace several times.  And I'm using GitHub CodeSpaces on github.com.

**Comment by jazzcat** [comment]
@douglasstarnes 

Same experience here after installing the extension (Codespaces on GitHub.com or vscode.dev) and logging into my MS account:

![two](https://github.com/microsoft/vscode-azureresourcegroups/assets/88254484/f868f120-e019-4ab4-b79b-7b53ce375452)

As it turns out, after logging into your MS account, you need to grant additional access to your Azure resources:

![one](https://github.com/microsoft/vscode-azureresourcegroups/assets/88254484/74e92a11-6822-448a-8b15-da3eba237f17)

Once that's been done, you should be all set:

![three](https://github.com/microsoft/vscode-azureresourcegroups/assets/88254484/82ab41b0-591c-4e5b-ab84-df2aa4ae57a4)

**Comment by alexweininger** [comment]
@jazzcat, thanks for reporting the workaround. I can confirm that we are working to fix having to do the additional grant.

Related to/duplicate of https://github.com/microsoft/vscode-azure-account/issues/797. I will leave this open for now until this is fixed.

**Comment by douglasstarnes** [comment]
I can confirm that this works.  It didn't use to be this way. Also, I would add that to see the **Grant Access to Microsoft for Azure Resources** option you must first click on **Select Subscription** in the Azure Tool pane and let that fail.  Then you will see the option to grant access and it will work.

**Comment by alexweininger** [comment]
I've verified that this issue is fixed in the current insiders release 1.80.0-insider. Try it out on [insiders.vscode.dev](https://insiders.vscode.dev)
